File: home/camisa.java
import java.util.*;

class Camiseta {
    String nome;
    String cor;
    String tamanho;

    public Camiseta(String nome, String cor, String tamanho) {
        this.nome = nome;
        this.cor = cor;
        this.tamanho = tamanho;
    }
}

public class camisa {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        List<Camiseta> camisetas = new ArrayList<>();
        boolean primeiroCaso = true;

        while (true) {
            int N = Integer.parseInt(scanner.nextLine().trim());
            if (N == 0) break;

            camisetas.clear(); // Limpa a lista para o próximo caso de teste

            for (int i = 0; i < N; i++) {
                String nome = scanner.nextLine().trim();
                String[] detalhes = scanner.nextLine().trim().split(" ");
                String cor = detalhes[0];
                String tamanho = detalhes[1];
                camisetas.add(new Camiseta(nome, cor, tamanho));
            }

            // Ordena as camisetas
            Collections.sort(camisetas, (a, b) -> {
                int corCompare = a.cor.compareTo(b.cor); // Ordem ascendente pela cor
                if (corCompare != 0) return corCompare;

                int tamanhoCompare = b.tamanho.compareTo(a.tamanho); // Ordem descendente pelo tamanho
                if (tamanhoCompare != 0) return tamanhoCompare;

                return a.nome.compareTo(b.nome); // Ordem ascendente pelo nome
            });

            // Imprime uma linha em branco entre os casos de teste, exceto no primeiro
            if (!primeiroCaso) {
                System.out.println();
            }
            primeiroCaso = false;

            // Imprime as camisetas ordenadas
            for (Camiseta camiseta : camisetas) {
                System.out.printf("%s %s %s%n", camiseta.cor, camiseta.tamanho, camiseta.nome);
            }
        }

        scanner.close();
    }
}

File: home/camisa.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct {
    char nome[51];
    char cor[10];
    char tamanho[2];
} Camiseta;

int compare(const void *a, const void *b) {
    Camiseta *c1 = (Camiseta *)a;
    Camiseta *c2 = (Camiseta *)b;

    // Comparar pela cor (ascendente)
    int corCompare = strcmp(c1->cor, c2->cor);
    if (corCompare != 0) return corCompare;

    // Comparar pelo tamanho (descendente)
    if (strcmp(c1->tamanho, c2->tamanho) != 0) {
        return strcmp(c2->tamanho, c1->tamanho); // inverte a comparação para ordem descendente
    }

    // Comparar pelo nome (ascendente)
    return strcmp(c1->nome, c2->nome);
}

int main() {
    int N;
    int primeiroCaso = 1;

    while (1) {
        scanf("%d", &N);
        if (N == 0) break;

        Camiseta *camisetas = malloc(N * sizeof(Camiseta));

        for (int i = 0; i < N; i++) {
            // Lê o nome do estudante
            getchar(); // Limpa o buffer do '\n'
            fgets(camisetas[i].nome, sizeof(camisetas[i].nome), stdin);
            camisetas[i].nome[strcspn(camisetas[i].nome, "\n")] = 0; // Remove o '\n'

            // Lê a cor e o tamanho
            scanf("%s %s", camisetas[i].cor, camisetas[i].tamanho);
        }

        // Ordena as camisetas
        qsort(camisetas, N, sizeof(Camiseta), compare);

        // Imprime uma linha em branco entre os casos de teste, exceto no primeiro
        if (!primeiroCaso) {
            printf("\n");
        }
        primeiroCaso = 0;

        // Imprime as camisetas ordenadas
        for (int i = 0; i < N; i++) {
            printf("%s %s %s\n", camisetas[i].cor, camisetas[i].tamanho, camisetas[i].nome);
        }

        free(camisetas);
    }

    return 0;
}
